pkg/kapp/cmd/app/deploy.go

```diff
@@ -215,35 +234,77 @@ func (o *DeployOptions) Run() error {
 	return nil
 }
 
+func (o *DeployOptions) newAndUsedGKs(newGKs []schema.GroupKind, app ctlapp.App) ([]schema.GroupKind, error) {
+	if o.ResourceTypesFlags.DisableGKScoping {
+		return []schema.GroupKind{}, nil
+	}
+
+	gksByGK := map[schema.GroupKind]struct{}{}
+	var uniqGKs []schema.GroupKind
+
+	usedGKs, err := app.UsedGKs()
+	if err != nil {
+		return nil, err
+	}
+
+	// Handle existing apps without cached GKs
+	// These apps can cache and scope to GKs in subsequent deploys
+	lastChange, err := app.LastChange()
```

Comment:
why do we need to check on last change?

Reply:
If LastChange is not nil, and there are no UsedGVs stored, it means that the app was created by an older version of kapp. (LastChange not being nil indicates that it has been deployed once).
Is there a better way of determining whether an app has been deployed by an older version of kapp?

Reply:
LastChange does not really indicate that app was deployed by older version of kapp. newer version of kapp also have LastChange (so do super old ones, right?).

i think what you probably want is to make usedGKs a pointer (*[]stuff) so that you can see a distinction between field being present or not.